File: testsystem-frontend/src/app/models/result-dto.ts
import {TestDto} from "./test-dto";
import {UserDto} from "./user-dto";
import {AssessmentDto} from "./assessment-dto";

export interface ResultDto {
  id: number;
  score: number;
  acceptResult: boolean;
  user: UserDto;
  test: TestDto;
  assessment: AssessmentDto | null;
}

File: testsystem-backend/src/main/java/com/underfit/testsystembackend/dto/ResultDto.java
package com.underfit.testsystembackend.dto;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

/**
 * DTO for {@link com.underfit.testsystembackend.entity.Result}
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class ResultDto implements Serializable {
    private Long id;
    private Integer score;
    private Boolean acceptResult;
    private UserDto user;
    private TestDto test;
    private AssessmentDto assessment;
}